### PR TITLE
 On branch AK-running-without-ticket-assignment-or-env-variables

### PR DESCRIPTION
**Pull request by Xtrems876**
Changes to be committed:
	modified:   index.ts - added logic for noTicketAssignment and errors for missing env variables
	modified:   package.json - added a command for 'npm run noTicketAssignment'
All below pertain to errors of lacking env variables
	modified:   src/middlewares/authMiddleware.ts
	modified:   src/services/helpscout/authenticationService.ts
	modified:   src/services/sendEmailService.ts
	modified:   src/services/zendesk/authenticationService.ts
	modified:   utilities/loggerTest.ts
	modified:   utilities/sendEmailsToAgents.ts
	modified:   utilities/updateDBShiftsFromCSV.ts

### CODE REVIEW

File: index.ts [hunk 1]
```diff
@@ -90,7 +92,12 @@ logger.saveLog({
 const job = new cron.CronJob('1/10 * 6-22 * * *',  async function () {
    assignNewTickets(logger); 
 });
-job.start();
+
+//Running without ticket assignment
+const noTicketAssignment = process.argv.includes('--noTicketAssignment');
+if (noTicketAssignment === true){
```
Comment (fszwedo) [inline index.ts:98]: I'd go with _if(process.argv.includes('--noTicketAssignment'))_ - one line less

File: index.ts [hunk 2]
```diff
@@ -53,6 +53,8 @@ const userController = new UserController(userService);
 const authService = new AuthService(userRepository, process.env.JWTPRIVATEKEY);
 const authController = new AuthController(authService, userService);
 
+if(!process.env.JWTPRIVATEKEY || !process.env.PORT){console.log("Either JWTPRIVATEKEY or PORT environment variable is not present!")}
+if(!process.env.MONGOLOGIN || !process.env.MONGOPW){throw new Error("Either MONGOLOGIN or MONGOPW environment variable is not present!")}
```
Comment (fszwedo) [inline index.ts:57]: if you have just one statement after _if_ then you can get rid of curly brackets, so e.g

if(!process.env.JWTPRIVATEKEY || !process.env.PORT) console.log("Either JWTPRIVATEKEY or PORT environment variable is not present!")

**Review by fszwedo — APPROVED** [review]
just two small change suggestions - please apply and then we can merge the changes

File: index.ts (hunk 1 above)
Comment (Xtrems876) [reply index.ts]: Done

File: index.ts (hunk 2 above)
Comment (Xtrems876) [reply index.ts]: Done for these 2 lines and lines in other files too.

**Pull request merged**